src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,37 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        try {
+            if (!executorService.awaitTermination(1, TimeUnit.SECONDS)) {
+                executorService.shutdown();
+            }
+        } catch (InterruptedException e) {
+            executorService.shutdown();
+            Thread.currentThread().interrupt();
```

Comment:
The shutdown logic seems incorrect. The awaitTermination method should be called after shutdown to await existing tasks to complete. The current logic is attempting to wait for termination before shutting down the executor, which will never succeed because the executor has not been instructed to shut down yet. The correct sequence is to first call shutdown, then awaitTermination, and if the awaitTermination returns false, that means not all tasks have completed and you should then call shutdownNow to cancel any remaining tasks.